GeneralsMD/Code/Tools/GenBIG/src/BIGFile.cpp

```diff
@@ -0,0 +1,109 @@
+#include "BIGFile.h"
+#include <iostream>
+#include <fstream>
+#include <cstring>
+#include <filesystem>
+
+BIGFile::BIGFile()
+{
+	init();
+}
+
+BIGFile::~BIGFile()
+{
+	if (m_archiveFile != nullptr) {
+		m_archiveFile->close();
+		delete m_archiveFile;
+		m_archiveFile = nullptr;
+	}
+	if (m_paths != nullptr) {
+		delete[] m_paths;
+		m_paths = nullptr;
+	}
+	if (m_outputDir != nullptr) {
+		delete[] m_outputDir;
+		m_outputDir = nullptr;
+	}
+	if (m_archiveName != nullptr) {
+		delete[] m_archiveName;
```

Comment:
Prefer to not use new/delete where so possible and use things like STL containers (`std::string` for strings) and smart pointers (`std::unique_ptr`, `std::shared_ptr`).

When interfacing with Engine, can also use `AsciiString` and `UnicodeString`.